lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GaussianProcessRegression.cxx

```diff
@@ -0,0 +1,203 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The class building gaussian process regression
+ *
+ *  Copyright 2005-2024 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/GaussianProcessRegression.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/KrigingEvaluation.hxx"
+#include "openturns/KrigingGradient.hxx"
+#include "openturns/CenteredFiniteDifferenceHessian.hxx"
+#include "openturns/GaussianProcessFitter.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(GaussianProcessRegression)
+
+static const Factory<GaussianProcessRegression> Factory_GaussianProcessRegression;
+
+
+/* Default constructor */
+GaussianProcessRegression::GaussianProcessRegression()
+  : MetaModelAlgorithm()
+  , covarianceModel_()
+  , basis_()
+  , beta_(0)
+  , gaussianProcessFitterResult_()
+  , result_()
+{
+  // Nothing to do
+}
+
+GaussianProcessRegression::GaussianProcessRegression(const GaussianProcessFitterResult &result)
```

Comment:
A similar constructor based on a GaussianProcessRegression result should be provided too

Reply:
???

Reply:
Maybe a new ctor like 
```cxx
GaussianProcessRegression::GaussianProcessRegression(const GaussianProcessFitterResult &result, 
                                                     const Sample & outputSample) 
```

what do you think both?